Lombiq.HelpfulLibraries.LinqToDb/Startup.cs

```diff
@@ -0,0 +1,64 @@
+using LinqToDB.AspNet;
+using LinqToDB.AspNet.Logging;
+using LinqToDB.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Modules;
+using System;
+using System.IO;
+using static Lombiq.HelpfulLibraries.LinqToDb.Constants.FeatureIds;
+using Configuration = LinqToDB.Common.Configuration;
+
+namespace Lombiq.HelpfulLibraries.LinqToDb
+{
+    [Feature(Default)]
+    public class Startup : StartupBase
+    {
+        private readonly IOptions<ShellOptions> _shellOptions;
+        private readonly ShellSettings _shellSettings;
+
+        public Startup(IOptions<ShellOptions> shellOptions, ShellSettings shellSettings)
+        {
+            _shellOptions = shellOptions;
+            _shellSettings = shellSettings;
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            switch (_shellSettings["DatabaseProvider"])
+            {
+                case "SqlConnection":
+                    services.AddLinqToDbContext<LinqToDbConnection>(
+                    (provider, options) =>
+                        options
+                        // Using explicit string instead of LinqToDB.ProviderName.SQLServer because the "System.Data.SqlClient"
+                        // provider will be used causing "Could not load type System.Data.SqlClient.SqlCommandBuilder"
+                        // exception. See: https://github.com/linq2db/linq2db/issues/2191#issuecomment-618450439
+                        .UseConnectionString("Microsoft.Data.SqlClient", _shellSettings["ConnectionString"])
+                        .UseDefaultLogging(provider),
+                    ServiceLifetime.Transient);
+                    break;
+                case "Sqlite":
+                    var option = _shellOptions.Value;
+                    var databaseFolder = Path.Combine(option.ShellsApplicationDataPath, option.ShellsContainerName, _shellSettings.Name);
+                    var databaseFile = Path.Combine(databaseFolder, "yessql.db");
+
+                    services.AddLinqToDbContext<LinqToDbConnection>(
+                    (provider, options) =>
+                        options
+                        .UseSQLite($"Data Source={databaseFile};Cache=Shared")
+                        .UseDefaultLogging(provider),
+                    ServiceLifetime.Transient);
+                    break;
+                default:
+                    throw new ArgumentException("Unknown database provider: " + _shellSettings["DatabaseProvider"]);

```

Comment:
E helyett nem lehet inkább a `DbConnection`-t használni az `IDbConnectionAccessor`-ból? Lásd: https://github.com/Lombiq/Finitive/pull/259#discussion_r541569160 Az eredeti, akkor (novemberi) kódból nekem úgy tűnik, ezt lehet egyszerűbben is.

Reply:
Azért választottam ezt mert a dokumentáció is ezt írja ASP.NET Core-ra:
[https://linq2db.github.io/articles/get-started/asp-dotnet-core/index.html](https://linq2db.github.io/articles/get-started/asp-dotnet-core/index.html)

Akkor azt mondod csináljam meg úgy ahogy már megvolt ezelőtt a DI-os szemlélet előtt?

Reply:
Szerintem annak akkor van értelme, ha nulláról kell a DB connectiont, de itt nem erről van szó.

Reply:
Oksa, jogos.